vfs.go

```diff
@@ -396,3 +399,107 @@ func (v *VFS) DirSize(uri string) (uint64, error) {
 
 	return uint64(cfsize), nil
 }
+
+// Close a file. This is flushes the buffered data into the file when the file
+// was opened in write (or append) mode. It is particularly important to be
+// called after S3 writes, as otherwise the writes will not take effect.
+func (v *VFSfh) Close() error {
+
+	ret := C.tiledb_vfs_close(v.context.tiledbContext, v.tiledbVFSfh)
+
+	if ret != C.TILEDB_OK {
+		return fmt.Errorf("Unknown error in VFS.Close: %s", v.context.LastError())
+	}
+
+	v.Free()
+	return nil
+}
+
+// Read part of a file
+func (v *VFSfh) Read(p []byte) (int, error) {
+	nbytes := uint64(len(p))
+
+	// If the size is empty, fetch it
+	if v.size == nil {
+		size, err := v.vfs.FileSize(v.uri)
+		if err != nil {
+			return 0, err
+		}
+		v.size = &size
+	}
+
+	// If the requested read size is beyond the limit, adjust bytes to read
+	if v.offset+nbytes > *v.size {
+		nbytes = *v.size - v.offset
+	}
+
+	if nbytes == 0 {
+		return 0, io.EOF
+	}
+
+	cbuffer := unsafe.Pointer(&p[0])
+	ret := C.tiledb_vfs_read(v.context.tiledbContext, v.tiledbVFSfh, C.uint64_t(v.offset), cbuffer, C.uint64_t(nbytes))
+
+	if ret != C.TILEDB_OK {
+		return 0, fmt.Errorf("Unknown error in VFS.Read: %s", v.context.LastError())
+	}
+
+	v.offset += nbytes
+	return int(nbytes), nil
+}
+
+// Write the contents of a buffer into a file. Note that this function only
+// appends data at the end of the file. If the file does not exist,
+// it will be created
+func (v *VFSfh) Write(bytes []byte) (int, error) {
+	cbuffer := C.CBytes(bytes)
+	ret := C.tiledb_vfs_write(v.context.tiledbContext, v.tiledbVFSfh, cbuffer, C.uint64_t(len(bytes)))
+
+	if ret != C.TILEDB_OK {
+		return 0, fmt.Errorf("Unknown error in VFS.Write: %s", v.context.LastError())
+	}
+
+	return len(bytes), nil
+}
+
+// Sync (flushes) a file.
+func (v *VFSfh) Sync() error {
+	ret := C.tiledb_vfs_sync(v.context.tiledbContext, v.tiledbVFSfh)
+
+	if ret != C.TILEDB_OK {
+		return fmt.Errorf("Unknown error in VFS.Sync: %s", v.context.LastError())
+	}
+
+	return nil
+}
+
+// Seek to an offset
+func (v *VFSfh) Seek(offset int64, whence int) (int64, error) {
+	absOffset := uint64(offset)
+	if offset <= 0 {
+		absOffset = uint64(-1 * offset)
+	}
+	var origin uint64
+
+	switch whence {
+	case io.SeekStart:
+		origin = 0
+		break
+	case io.SeekCurrent:
+		origin = v.offset
+		break
+	case io.SeekEnd:
+		origin = *v.size
```

Comment:
Are we sure that `v.size` is set?

Reply:
Also origin is 0 after `var origin uint64` so `SeekEnd` will be 0 always right?

Reply:
Thanks, adjusted